RecoTracker/FinalTrackSelectors/plugins/TrackSelectorByRegion.cc

```diff
@@ -0,0 +1,85 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/Common/interface/OwnVector.h"
+#include "RecoTracker/TkTrackingRegions/interface/TrackingRegion.h"
+#include "RecoTracker/TkTrackingRegions/interface/RectangularEtaPhiTrackingRegion.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include <vector>
+#include <memory>
+
+namespace {
+  class TrackSelectorByRegion final : public edm::global::EDProducer<> {
+  public:
+    explicit TrackSelectorByRegion(const edm::ParameterSet& conf)
+        : tracksToken_(consumes<reco::TrackCollection>(conf.getParameter<edm::InputTag>("tracks"))) {
+      inputTrkRegionToken_ = consumes<edm::OwnVector<TrackingRegion>>(conf.getParameter<edm::InputTag>("inputROI"));
+      produce_collection = conf.getParameter<bool>("produceTrackCollection");
+      if(produce_collection) produces<reco::TrackCollection>();
+      produces<std::vector<bool>>();
+    }
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
+      desc.add<edm::InputTag>("tracks", edm::InputTag("hltPixelTracks"));
+      desc.add<edm::InputTag>("inputROI", edm::InputTag(""));
+      desc.add<bool>("produceTrackCollection", true);
+      descriptions.add("trackSelectorByRegion", desc);
+    }
+
+  private:
+    using MaskCollection = std::vector<bool>;
+
+    void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup&) const override {
+      // products
+      auto mask = std::make_unique<MaskCollection>();                  // mask w/ the same size of the input collection
+      auto output_tracks = std::make_unique<reco::TrackCollection>();  // selected output collection
+
+      auto regionsHandle = iEvent.getHandle(inputTrkRegionToken_);
+
+      auto tracksHandle = iEvent.getHandle(tracksToken_);
+
+      if (tracksHandle.isValid()) {
```

Comment:
I would remove the `isValid()` check; otherwise this module sill silently ignore any errors in the configuration.
If the handle is not valid it's OK to get a run time error.